crates/pep440-rs/src/version.rs

```diff
@@ -1415,6 +1415,20 @@ impl std::fmt::Display for Prerelease {
     }
 }
 
+impl FromStr for Prerelease {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let version = Version::from_str(s).map_err(|err| err.to_string())?;
+        match version.pre() {
+            Some(pre) => Ok(pre),
+            None => Err(format!(
+                "unable to parse pre-release version from the given input: {s}"
+            )),
+        }
+    }
+}
```

Comment:
This feels a little weird. I don't think this function should support leading content. Maybe we should just implement this directly in the relevant version request parsing section instead of implementing `FromStr`? cc @BurntSushi

Reply:
Yep, I now removed this.